model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java

```diff
@@ -897,7 +897,7 @@ protected void ensureEmailConstraint(List<UserEntity> users, RealmModel realm) {
     }
 
     private Predicate[] getSearchOptionPredicateArray(String value, CriteriaBuilder builder, From<?, UserEntity> from) {
-        value = value.toLowerCase();
+        value = value.toLowerCase().replace("%", "\\%").replace("_", "\\_");
```

Comment:
I gave it a thought I think first all existing `\` should be escaped first before escaping the special characters.